### PR TITLE
fix: setup.py --all only put local driver if arch matches

### PR DESCRIPTION
**Pull request by mxschmitt**
Before this CL if you have used setup.py -all it resulted in that the local driver got overridden by the last matching downloaded driver. Only Linux or MacOS this would be then the wrong architecture in most cases.

drive-by: had to rename it a bit, because it conflicted otherwise with the imported platform module.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -59,57 +60,73 @@ def run(self) -> None:
...
                 "win32": [
                     {
                         "zip_name": "win32_x64",
                         "wheels": ["win32.whl", "win_amd64.whl"],
+                        "machine": "amd64",
                     }
                 ],
             }
-            platforms = [*platform_map.values()]
+            pw_platforms = pw_platform_map.items()
```
Comment (pavelfeldman) [inline setup.py:101]: Why did this change?

File: setup.py [hunk 2]
```diff
@@ -119,12 +136,15 @@ def run(self) -> None:
         base_wheel_location = glob.glob(os.path.join(self.dist_dir, "*.whl"))[0]
         without_platform = base_wheel_location[:-7]
 
-        for platform in platforms:
-            for arch in platform:
+        for pw_architecture, pw_platform in pw_platforms:
+            for arch in pw_platform:
```
Comment (pavelfeldman) [inline setup.py:140]: Looks like "machine" is "arch", and this was "platform"?

File: setup.py [hunk 3]
```diff
@@ -119,12 +136,15 @@ def run(self) -> None:
...
                 zip_file = f"driver/playwright-{driver_version}-{arch['zip_name']}.zip"
                 with zipfile.ZipFile(zip_file, "r") as zip:
                     extractall(zip, f"driver/{arch['zip_name']}")
-                if platform_map[sys.platform][0] in platform:
+                if (
+                    arch["machine"] == platform.machine().lower()
+                    and pw_architecture == sys.platform
```
Comment (pavelfeldman) [inline setup.py:146]: Confused, you can't compare platform to architecture.

**Review by kumaraditya303 — APPROVED** [review]

File: setup.py [hunk 4]
```diff
@@ -14,13 +14,16 @@
...
+import platform
 import shutil
 import subprocess
 import sys
 import zipfile
 from pathlib import Path
+from typing import List
 
 from setuptools import find_packages, setup
+from typing_extensions import TypedDict
```
Comment (kumaraditya303) [inline setup.py:26]: This would break the build when this isn't installed.

File: setup.py [hunk 5]
```diff
@@ -57,96 +68,113 @@ def run(self) -> None:
...
-                    url = url + zip_file
-                    print(f"Fetching {url}")
-                    # Don't replace this with urllib - Python won't have certificates to do SSL on all platforms.
-                    subprocess.check_call(["curl", url, "-o", "driver/" + zip_file])
-        base_wheel_location = glob.glob(os.path.join(self.dist_dir, "*.whl"))[0]
+            wheel_bundles = list(
+                filter(
+                    lambda bundle: bundle["machine"] == platform.machine()
+                    and bundle["platform"] == sys.platform
+                    and bundle["dev_build"] is True,
```
Comment (kumaraditya303) [inline setup.py:131]:
```suggestion
                    and bundle["dev_build"],
```

File: setup.py [hunk 6]
```diff
@@ -57,96 +68,113 @@ def run(self) -> None:
...
+            zip_file = (
+                f"driver/playwright-{driver_version}-{wheel_bundle['zip_name']}.zip"
+            )
+            with zipfile.ZipFile(zip_file, "r") as zip:
+                extractall(zip, f"driver/{wheel_bundle['zip_name']}")
+            # Driver for local development
+            if (
+                wheel_bundle["machine"] == platform.machine().lower()
+                and wheel_bundle["platform"] == sys.platform
+                and wheel_bundle["dev_build"] is True
```
Comment (kumaraditya303) [inline setup.py:161]:
```suggestion
                and wheel_bundle["dev_build"]
```

**Review by kumaraditya303 — CHANGES_REQUESTED** [review]

File: setup.py (hunk 4 above)
Comment (mxschmitt) [reply setup.py]: reverted

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**